### PR TITLE
Made .index inside Maven 2 Repositories browsable #548

### PR DESCRIPTION
**Pull request by sanket407**
#548  
Made .index browsable inside Maven 2 Repositories !

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseArtifactController.java [hunk 1]
```diff
@@ -96,10 +97,12 @@ protected boolean probeForDirectoryListing(Repository repository,
...
 
         File file = new File(dir);
-
-        // Do not allow .index and .trash directories (or any other directory starting with ".") to be browseable.
+        
+        // For Maven 2 Repositories only allow .index to be browsable directly by admins
+        // For others do not allow .index and .trash directories (or any other directory starting with ".") to be browseable.
         // NB: Files will still be downloadable.
-        if (!file.isHidden() && !path.startsWith(".") && !path.contains("/."))
+        if ((repository.getLayout().equals(RepositoryLayoutEnum.MAVEN_2.getLayout()) && path.startsWith(".index"))
```
Comment (fuss86) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseArtifactController.java]: Would you mind extracting a template method here and override it in the `org.carlspring.strongbox.controllers.maven.MavenArtifactController` ? That way we'll leave the `org.carlspring.strongbox.controllers.BaseArtifactController` more generic without any repository layout specific information.

**Comment by fuss86** [comment]
@carlspring, one comment here. Other than that, it looks good and works as expected. Thanks @sanket407 !

**Comment by carlspring** [comment]
This sounds like a very good idea! Thanks!

**Comment by sanket407** [comment]
@fuss86 I have added template method. Please review,

**Comment by fuss86** [comment]
> @fuss86 I have added template method. Please review,

I was thinking about adding a new template method to the existing `BaseArtifactController.probeForDirectoryListing` method.

I meant replacing `BaseArtifactController.java`:

```
if (!file.isHidden() && !path.startsWith(".") && !path.contains("/."))
```

by:

```
if (isAllowedForDirectoryListing(file, path))
```

and add default implementation inside `BaseArtifactController.java`:

```
protected boolean isAllowedForDirectoryListing(File file, String path) {
return (!file.isHidden() && !path.startsWith(".") && !path.contains("/."))
}
```

and inside `MavenArtifactController.java`:

```
protected boolean isAllowedForDirectoryListing(File file, String path) {
if (path.startsWith(".index") || (super.isAllowedForDirectoryListing(file, path)))
}
```

This way you don't have to repeat the same code inside `MavenArtifactController.java`.

**Comment by sanket407** [comment]
@fuss86  Done ! That was indeed nice way of doing it . Thanks !

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java [hunk 2]
```diff
@@ -4,12 +4,12 @@
 import org.carlspring.strongbox.storage.ArtifactStorageException;
 import org.carlspring.strongbox.storage.Storage;
 import org.carlspring.strongbox.storage.repository.Repository;
-
+import org.carlspring.strongbox.storage.repository.RepositoryLayoutEnum;
```
Comment (fuss86) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java]: unnecessary import added :)

**Comment by fuss86** [comment]
Looks good

**Comment by carlspring** [comment]
Thanks, guys! I've removed the unused imports!

**Pull request merged**